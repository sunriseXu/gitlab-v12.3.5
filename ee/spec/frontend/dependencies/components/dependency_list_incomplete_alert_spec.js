import { createLocalVue, shallowMount } from '@vue/test-utils';
import DependencyListAlert from 'ee/dependencies/components/dependency_list_alert.vue';
import DependencyListIncompleteAlert from 'ee/dependencies/components/dependency_list_incomplete_alert.vue';

describe('DependencyListIncompleteAlert component', () => {
  let wrapper;

  const factory = (options = {}) => {
    const localVue = createLocalVue();

    wrapper = shallowMount(localVue.extend(DependencyListIncompleteAlert), {
      localVue,
      sync: false,
      ...options,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('matches the snapshot', () => {
    factory();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('when the generic alert component emits a close event', () => {
    let closeListenerSpy;

    beforeEach(() => {
      closeListenerSpy = jest.fn();

      factory({
        listeners: {
          close: closeListenerSpy,
        },
      });

      wrapper.find(DependencyListAlert).vm.$emit('close');
    });

    it('calls the given listener', () => {
      expect(closeListenerSpy).toHaveBeenCalledTimes(1);
    });
  });
});
