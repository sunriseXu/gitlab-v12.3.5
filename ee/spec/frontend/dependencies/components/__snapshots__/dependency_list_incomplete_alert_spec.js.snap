// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyListIncompleteAlert component matches the snapshot 1`] = `
<dependencylistalert-stub
  headertext="Unsupported file(s) detected"
  type="warning"
>
  <p>
    
    One or more of your dependency files are not supported, and the dependency list may be incomplete. Below is a list of supported file types.
  
  </p>
   
  <ul>
    <li>
      package-lock.json
    </li>
    <li>
      composer.lock
    </li>
    <li>
      Gemfile.lock
    </li>
    <li>
      gems.locked
    </li>
    <li>
      yarn.lock
    </li>
    <li>
      requirements.txt
    </li>
    <li>
      pom.xml
    </li>
  </ul>
</dependencylistalert-stub>
`;
